### ISSUE TITLE
[prefer-optional-chain] check negated statements with `||`

### ISSUE THREAD

**Issue by bogdan** [typescript-eslint/typescript-eslint#1988]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-optional-chain": "error",
  }
}
```

```TS
const a = {b: 1};
const c = !a || !a.b;
```

**Expected Result**

```
@typescript-eslint/prefer-optional-chain: Prefer using an optional chain expression instead, as it's more concise and easier to read. 
```

Code is fixed to:

```
const a = {b: 1};
const c = !a?.b;
```

**Actual Result**

No error

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.28.0` |
| `@typescript-eslint/parser`        | `2.28.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |

**Comment by bradzacher** [comment]
Fixed in #5594
![image](https://user-images.githubusercontent.com/7462525/215768559-de663f18-1233-429c-97ad-5ff3d67c7144.png)
